internal/cluster/distribution/pke/pkeaws/pkeawsadapter/image_selector.go

```diff
@@ -84,5 +84,18 @@ func (s CloudinfoImageSelector) SelectImage(ctx context.Context, criteria pkeaws
 		)
 	}
 
+	// As a result of a bug in cloudinfo,
+	// the returned item might be empty
+	// See https://github.com/banzaicloud/cloudinfo/pull/356
+	if images[0].Name == "" {
+		return "", errors.WithDetails(
+			errors.WithStack(pkeaws.ImageNotFoundError),
+			"cloudProvider", cloudProvider,
+			"service", serviceName,
+			"region", criteria.Region,
+			"getImagesOpts", opts,
+		)
+	}
+
```

Comment:
**Optional**: we **MAY** consider adding the condition (`images[0].Name == ""`) to the previous `if` statement as both branches are supposed to return the same values and that way code duplication can be avoided.

I think the temporal state of this case and corresponding external module comment is more explicit/visible this way which is a plus, but I don't like the duplicated branch.

I guess it is opinionated, so it's totally up for you to choose, I just wanted to remark it if it was not deliberately done this way.

Reply:
I deliberately added a separate branch, because it's easier to clean up once the bug is resolved.